dpctl/_sycl_device.pyx

```diff
@@ -473,11 +360,125 @@ cdef class SyclDevice(_SyclDevice):
             DPCTLDeviceSelector_Delete(DSRef)
         return score
 
+    @property
+    def is_accelerator(self):
+        """ Returns True if the SyclDevice instance is a SYCL accelerator
+        device.
+
+        Returns:
+            bool: True if the SyclDevice is a SYCL accelerator device,
+            else False.
+        """
+        return DPCTLDevice_IsAccelerator(self._device_ref)
+
+    @property
+    def is_cpu(self):
+        """ Returns True if the SyclDevice instance is a SYCL CPU device.
+
+        Returns:
+            bool: True if the SyclDevice is a SYCL CPU device, else False.
+        """
+        return DPCTLDevice_IsCPU(self._device_ref)
+
+    @property
+    def is_gpu(self):
+        """ Returns True if the SyclDevice instance is a SYCL GPU device.
+
+        Returns:
+            bool: True if the SyclDevice is a SYCL GPU device, else False.
+        """
+        return DPCTLDevice_IsGPU(self._device_ref)
+
+    @property
+    def is_host(self):
+        """ Returns True if the SyclDevice instance is a SYCL host device.
+
+        Returns:
+            bool: True if the SyclDevice is a SYCL host device, else False.
+        """
+        return DPCTLDevice_IsHost(self._device_ref)
+
+    @property
+    def max_work_item_dims(self):
+        """ Returns the maximum dimensions that specify
+            the global and local work-item IDs used by the
+            data parallel execution model. The cb
+            value is 3 if this SYCL device is not of device
+            type ``info::device_type::custom``.
+        """
+        cdef uint32_t max_work_item_dims = 0
+        max_work_item_dims = DPCTLDevice_GetMaxWorkItemDims(self._device_ref)
+        return max_work_item_dims
+
+    @property
+    def max_work_item_sizes(self):
+        """ Returns the maximum number of work-items
+            that are permitted in each dimension of the
+            work-group of the nd_range. The minimum
+            value is (1; 1; 1) for devices that are not of
+            device type ``info::device_type::custom``.
+        """
+        max_work_item_sizes = []
+        for n in range(3):
+            max_work_item_sizes.append(self._max_work_item_sizes[n])
+        return tuple(max_work_item_sizes)
```

Comment:
Cython will generate more efficient code from doing 
```python
    return tuple(
        self._max_work_item_sizes[0],
        self._max_work_item_sizes[1],
        self._max_work_item_sizes[2]
    )
```

This avoids needless list temporary.

Reply:
I made the changes, but am now getting this error

```
Traceback (most recent call last):
  File "/home/diptorupd/Desktop/devel/dpctl/dpctl/tests/test_sycl_queue.py", line 326, in test_valid_filter_selectors
    check(device)
  File "/home/diptorupd/Desktop/devel/dpctl/dpctl/tests/test_sycl_queue.py", line 66, in check_get_max_work_item_sizes
    max_work_item_sizes = device.max_work_item_sizes
  File "dpctl/_sycl_device.pyx", line 421, in dpctl._sycl_device.SyclDevice.max_work_item_sizes.__get__
    return tuple(
TypeError: tuple expected at most 1 argument, got 3

```